src/transformers/trainer.py

```diff
@@ -1680,7 +1680,9 @@ def _inner_training_loop(
         logger.info("***** Running training *****")
         logger.info(f"  Num examples = {num_examples:,}")
         logger.info(f"  Num Epochs = {num_train_epochs:,}")
-        logger.info(f"  Instantaneous batch size per device = {self._train_batch_size:,}")
+        logger.info(f"  Instantaneous batch size per device = {self.args.per_device_train_batch_size:,}")
+        if self.args.per_device_train_batch_size != self._train_batch_size:
+            logger.info(f"  Training with DataParallel so batch size has been adjusted to: {self._train_batch_size:,}")
         logger.info(f"  Total train batch size (w. parallel, distributed & accumulation) = {total_train_batch_size:,}")
```

Comment:
Thanks for care! #24345 seems resolved.

---

As I said https://github.com/huggingface/transformers/issues/24345#issuecomment-1601997057, I will not use DP anymore, but it seems to have bug around this logging part in case of DP.

e.g.) `total_train_batch_size` seems be larger than expected real value..?

May use 
```py
        total_train_batch_size = self.args.per_device_train_batch_size * args.gradient_accumulation_steps * args.world_size
```
at least in case of DP, instead of
https://github.com/huggingface/transformers/blob/8f093fb799246f7dd9104ff44728da0c53a9f67a/src/transformers/trainer.py#L1665

..?


I'm not sure about this issue and my suggestion (as I'm not familiar yet with trainer core/internal) so by now I will not report any issue or make a PR, but you guys may care about this.

I would try to make new issue or PR if I have enough time for check :smile: